_posts/2022-06-15-ejb-over-https.adoc

```diff
@@ -0,0 +1,340 @@
+        </credentials>
+        <sasl-mechanism-selector selector="SCRAM-SHA-512-PLUS"/>
+        <providers>
+            <use-service-loader/>
+        </providers>
+    </configuration>
+</authentication-configurations>
+----
+
+Below this is the configuration for the TLS connection, under the `<ssl-contexts>` tag. The client specifies both the
+trust and key store to use for the connection. It also specifies the cipher suites and protocols it supports.
+
+[source,xml]
+.wildfly-config.xml
+----
+<ssl-contexts>
+    <ssl-context name="example-ssl">
+        <key-store-ssl-certificate key-store-name="sslClientKeyStore" alias="client">
+            <key-store-clear-password password="clientKeySecret"/>
+        </key-store-ssl-certificate>
+        <trust-store key-store-name="sslClientTrustStore"/>
+        <cipher-suite names="TLS_AES_128_GCM_SHA256" selector="DEFAULT"/>
+        <protocol names="TLSv1.3 TLSv1.2"/>
+    </ssl-context>
+</ssl-contexts>
+----
+
+Elytron does not enable TLS 1.3 by default, so we manually do so using the `<cipher-suite names>` attribute. For more
+information on configuring TLS 1.3, take a look at this
+https://wildfly-security.github.io/wildfly-elytron/blog/tls-13-with-wildfly/[blog post].
+
+Finally, under the `<jboss-ejb-client>` tag, the client will attempt to access the EJB on localhost port 8443, the
+WildFly server default for HTTP over TLS connections.
+
+[source,xml]
+.wildfly-config.xml
+----
+<jboss-ejb-client xmlns="urn:jboss:wildfly-client-ejb:3.2">
+    <connections>
+        <connection uri="remote+https://127.0.0.1:8443"/>
+    </connections>
+</jboss-ejb-client>
+----
+
+== Configuring the Security Domain
+
+We will now configure the server to require a login. Returning to the terminal with the management CLI, we create a
+FileSystem realm called `sslFsRealm` in the server configuration directory:
+
+[source]
+----
+/subsystem=elytron/filesystem-realm=sslFsRealm:add(path=sslFsRealmUsers, relative-to=jboss.server.config.dir)
+----
+
+Then, we add the `example_user` identity to the realm, and set a `groups` attribute with the `guest` value :
+
+[source]
+----
+/subsystem=elytron/filesystem-realm=sslFsRealm:add-identity(identity=example_user)
+
+/subsystem=elytron/filesystem-realm=sslFsRealm:set-password(identity=example_user, clear={password=examplePwd1!})
+
+/subsystem=elytron/filesystem-realm=sslFsRealm:add-identity-attribute(identity=example_user, name=groups, value=[guest])
+----
+
+Next, we will configure the security domain to use the filesystem realm. We'll use the included `groups-to-roles` role
+decoder to convert the `groups` attribute to a `guest` role:
+
+[source]
+----
+/subsystem=elytron/security-domain=sslFsSD:add(realms=[{realm=sslFsRealm,role-decoder=groups-to-roles}],default-realm=sslFsRealm,permission-mapper=default-permission-mapper)
+----
+
+Afterwards, we add a mapping to the security domain into the `ejb3` subsystem, for setting access based on roles:
```

Comment:
s/for setting access based on roles/for securing the EJB

(this is for both authentication and authorization)